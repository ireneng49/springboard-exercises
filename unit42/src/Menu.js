import React from "react";
import { Link } from "react-router-dom";
import "./FoodMenu.css";
import {
  Card,
  CardBody,
  CardTitle,
  CardText,
  ListGroup,
  ListGroupItem
} from "reactstrap";

function Menu({ items, title }) {

  //Based on title passed down, it will either create drink links, or snack links
  let itemType = (title==="Snacks") ? "snacks" : "drinks";

  return (
    <section className="col-md-4">
      <Card>
        <CardBody>
          <CardTitle className="font-weight-bold text-center">
            {itemType.charAt(0).toUpperCase() + itemType.slice(1)} Menu
          </CardTitle>
          <CardText>
            Some quick example text to build on the card title and make up the
            bulk of the card's content.
          </CardText>
          <ListGroup>
            {items.map(item => (
              <Link to={`/${itemType}/${item.id}`} key={item.id}>
                <ListGroupItem>{item.name}</ListGroupItem>
              </Link>
            ))}
          </ListGroup>
        </CardBody>
      </Card>
    </section>
  );
}

export default Menu;
