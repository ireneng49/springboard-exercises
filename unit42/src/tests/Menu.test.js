import React from "react";
import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import '@testing-library/jest-dom/extend-expect';
import Menu from "../Menu";

let snacks = [
    {
      "id": "nachos",
      "name": "Nachos",
      "description": "An American classic!",
      "recipe": "Cover expensive, organic tortilla chips with Cheez Whiz.",
      "serve": "Serve in a hand-thrown ceramic bowl, garnished with canned black olives"
    },
    {
      "id": "hummus",
      "name": "Hummus",
      "description": "Sure to impress your vegan friends!",
      "recipe": "Purchase one container of hummus.",
      "serve": "Place unceremoniously on the table, along with pita bread."
    },
    {
      "id": "arugula-and-walnut-salad",
      "name": "Arugula and Walnut Salad",
      "description": "Tart and delicious.",
      "recipe": "Mix arugula, toasted walnuts, and thinly-sliced Parmesan cheese. Dress with lemon and olive oil.",
      "serve": "Place on tiny, precious little plates."
    }
  ];

it("matches the snapshot", function() {
    const {asFragment} = render(
    <MemoryRouter initialEntries={["/snacks"]}>   
        <Menu items={snacks} title="Snacks"/>
    </MemoryRouter>
    
    )
    expect(asFragment()).toMatchSnapshot();
})

it("should render", function(){
    render(
    <MemoryRouter initialEntries={["/snacks"]}>   
        <Menu items={snacks} title="Snacks"/>
    </MemoryRouter>    
    );
})

it("should contain some link elements", function() {
    const {getByText} = render(
    <MemoryRouter initialEntries={["/snacks"]}>   
        <Menu items={snacks} title="Snacks"/>
    </MemoryRouter>
    )
    let snacksLink = getByText('Snacks Menu');
    expect(snacksLink).toBeInTheDocument()
    let nachos = getByText('Nachos')
    expect(nachos).toBeInTheDocument()

    let hummus = getByText('Hummus')
    expect(hummus).toBeInTheDocument()

    let salad = getByText('Arugula and Walnut Salad')
    expect(salad).toBeInTheDocument()
})