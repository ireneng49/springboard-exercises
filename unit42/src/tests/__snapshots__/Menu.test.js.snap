// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<DocumentFragment>
  <section
    class="col-md-4"
  >
    <div
      class="card"
    >
      <div
        class="card-body"
      >
        <h5
          class="font-weight-bold text-center card-title"
        >
          Snacks Menu
        </h5>
        <p
          class="card-text"
        >
          Some quick example text to build on the card title and make up the bulk of the card's content.
        </p>
        <ul
          class="list-group"
        >
          <a
            href="/snacks/nachos"
          >
            <li
              class="list-group-item"
            >
              Nachos
            </li>
          </a>
          <a
            href="/snacks/hummus"
          >
            <li
              class="list-group-item"
            >
              Hummus
            </li>
          </a>
          <a
            href="/snacks/arugula-and-walnut-salad"
          >
            <li
              class="list-group-item"
            >
              Arugula and Walnut Salad
            </li>
          </a>
        </ul>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
